Fix SMILES normalization

edgarsmdn opened the pull request:
Hi!

I noticed that before you had an exception for DMSO when standardizing SMILES. In fact this problem is present for any molecule with the sulfynil group (S=O). For some reason `rdMolStandardize` changes the representation of this group to the zwitterionic form ([S+][O-]). 

I added a function that detects the zwitterionic form and replace it with the original one. 
I also included a new `test` to exemplified the original problem and solution with several examples where this happens.

I am not aware of a good reason for making the zwitterionic form the standard, I'll also ask to the rdkit people why they choose the zwitterionic form to be the standard...

edgarsmdn commented:
rdkit discussion here: https://github.com/rdkit/rdkit/discussions/7973#discussion-7394554

simonmb commented:
I am wondering if its not better to solve this using a SMARTS pattern and use GetSubstructureMatches. In any case, I would wait some days to see what the guys from rdkit say to not fix something that might have other implications.

simonmb commented:
Also more common names would be sulfinyl or thionyl.

simonmb commented:
this might be interesting: https://greglandrum.github.io/rdkit-blog/posts/2024-02-23-custom-transformations-and-logging.html

simonmb commented:
Also this might be related: https://github.com/rdkit/rdkit/issues/3460

edgarsmdn commented:
Hi Simon, I modified the normalization function of the sulfynil group as you suggested, using `SMARTS` and `GetSubstructMatches`. I have also added a little notice message in the README regarding [this](https://github.com/MoleculeResolver/molecule-resolver/issues/19).
